packages/conferer/src/Conferer.hs

```diff
@@ -52,12 +55,25 @@ import Conferer.Config (Defaults)
 --   but malformed somehow (@"abc"@ as an Int) but that depends on the 'FromConfig'
 --   implementation for the type.
 fetch :: forall a. (FromConfig a, Typeable a, DefaultConfig a) => Config -> IO a
-fetch c = fetchFromRootConfigWithDefault c configDef
+fetch c = fetch' c configDef
 
 -- | Same as 'fetch' but it accepts the default as a parameter instead of using
 --   the default from 'configDef'
 fetch' :: forall a. (FromConfig a, Typeable a) => Config -> a -> IO a
-fetch' = fetchFromRootConfigWithDefault
+fetch' c a = do
+  asTopLevel $ fetchFromRootConfigWithDefault c a
+
+-- | Given an IO action that run it but if it throws a Conferer related excpetion
+-- pretty print it and exit the program with failure.
+asTopLevel :: IO a -> IO a
+asTopLevel action =
+  action
+     `catch` (\(e :: MissingRequiredKey) -> do
+       putStrLn $ displayException e
+       exitFailure)
+     `catch` (\(e :: ConfigParsingError) -> do
+       putStrLn $ displayException e
+       exitFailure)
```

Comment:
Not really, since the `catch` must be explicitly typed, and also I expect this to change quite a bit in the not so distant future when we add a way to introspect the possible values